lib/config.docker.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
```

Comment:
Isn't this self-defeating to make the values globally available twice?

Reply:
This is only easy option.

You can't set static properties dynamically with function call so I have to first save them into constants.

But it may also work with normal variables instead of constants.

Reply:
> This is only easy option.

I think putting the argument in the connection utility fits better with a heroku + docker (12-factor) deployment.

Reply:
> I think putting the argument in the connection utility fits better with a heroku + docker (12-factor) deployment.

Can you explain this a bit more?

Reply:
So instead of having config be the source for all things, you could instead build an interface and have multiple implementations that fulfil that interface to provide details.

A Lighter weight option that does similar would be using null coalesce operator, leaving the config as null, and `Config::$value ?? ENV['ENV_VAR1'] ?? ENV['ENV_VAR2'];`

http://sandbox.onlinephpfunctions.com/code/22d382de6f0128a63d11ba37b79a4c7d3faad57a is a trivial example

Reply:
It is now setting properties directly from env vars. It doesn't need null coalesce operator, because env vars need to be set anyway.